middleware.ts

```diff
@@ -0,0 +1,12 @@
+import type { NextRequest } from 'next/server';
+import { urlLocaleMiddleware } from '~/url-locale-middleware';
+
+export const middleware = (request: NextRequest) => {
+    return urlLocaleMiddleware(request);
+}
+
+export const config = {
+    matcher: [
+        '/((?!api|_next/static).*)',
```

Comment:
I also suggest you include that in the matcher. Next will replace all of the image URLs with optimized ones (starting from _next/image). Also, you definitely want to receive your favicon, so disable the language check on that one too.
```suggestion
        '/((?!api|_next/static|_next/image|favicon.ico).*)',
```